### ISSUE TITLE
Collection view search

### ISSUE THREAD

**Issue by adomasven** [zotero/zotero#1073]
Collection view search is requested fairly frequently with some users managing over [1000 collections](https://forums.zotero.org/discussion/60998/search-collections/).

UI Changes:

1. Move gear icon 
![image](https://cloud.githubusercontent.com/assets/5899315/17546604/adccd24e-5eeb-11e6-9ef0-ed2e21bd4ff5.png) next to the sync icon on the right 
![image](https://cloud.githubusercontent.com/assets/5899315/17546619/bce00d0a-5eeb-11e6-94fc-3e97c33438a2.png)
2. Add a search-box where the gear icon used to be

Interaction:

1. When typing into textbox, filter all collections (expanded and collapsed). Display in a tree hierarchy, highlighting matches. Siblings that do not match the search criteria are hidden.
2. Matching collections are not auto-selected and middle-pane displays last selected view. On selection, middle view changes.
3. When textbox is empty, restore previous collapse/expand order. If a collection was selected during the search, keep that expanded too.

**Comment by dstillman** [comment]
Rather than another search box that's visible by default, this would be a button (e.g., magnifying glass) that slid a search box into view.

This is all likely premised on a more major UI redesign that changed the toolbars to make it clearer that each section was part of the pane below it (but also kept reasonable treatment of global button like Sync and Actions). I don't think we want to add another search button in the current design.

**Comment by rmzelle** [comment]
How about putting a search box underneath the toolbar, at the top of the collection listing? E.g.:

![image](https://cloud.githubusercontent.com/assets/77951/17564685/4429dca8-5f02-11e6-8493-4c22ddeb4315.png)

**Comment by KarlHegbloom** [comment]
I like it. I could also use view histories and different remembered sort
and column layouts per subcollection or search -> search result.

**Comment by dstillman** [comment]
No, we don't want that as a permanent UI element — that would make three visible search boxes in the UI, and it would take up space even when unused, which for probably most people would be nearly all the time. That's roughly what it would look like when activated, though.

**Comment by gregoryfu** [comment]
I just wanted to put in a vote for this feature.  Thanks.

**Comment by syrte** [comment]
> f the collection li

I like this one.
